### PR TITLE
Implementing exceptions and its use

### LINKED ISSUE #76: Exceptions handling
Exception and uses

- [x] Implement exceptions
- [x] Test exceptions

### PR DESCRIPTION
**Pull request by nanrehd**
Including exceptions there is also a major change for tests:

- Updated 'build.gradle' so that tests are executed.
- Most tests have been updated and new were added to check the functionality of exceptions.

Note: Due to an environment bug(not solved yet) there might be changes done before merging regarding an especific exception.

### CODE REVIEW

File: src/main/java/org/jalasoft/moi/model/exceptions/ProcessIDException.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *  Copyright (c) 2020 Jalasoft.
+ *
+ *  This software is the confidential and proprietary information of Jalasoft.
+ *  ("Confidential Information"). You shall not disclose such Confidential
+ *  Information and shall use it only in accordance with the terms of the
+ *  license agreement you entered into with Jalasoft.
+ */
+
+package org.jalasoft.moi.model.exceptions;
+
```
Comment (Snip-77) [inline src/main/java/org/jalasoft/moi/model/exceptions/ProcessIDException.java:11]: Delete one space

File: src/test/java/org/jalasoft/moi/model/exceptions/InputParametersExceptionTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import org.junit.Test;
+
+public class InputParametersExceptionTest {
+
+    @Test(expected = InputParametersException.class)
+    public void inputParametersExceptionNullTest() throws InputParametersException, ResultException {
+        Executer executer = new Executer(new ProcessCache());
+        executer.processAnswer(null);
+    }
+}
```
Comment (Snip-77) [inline src/test/java/org/jalasoft/moi/model/exceptions/InputParametersExceptionTest.java]: Missing one additional space

File: src/test/java/org/jalasoft/moi/model/exceptions/ParametersExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            handler.runProgram(params);
+        });
+
+        String expected = "Invalid or Null parameters gere generated.";
+        String actual = exception.getMessage();
+
+        assertEquals(expected, actual);
+    }
+
+}
```
Comment (Snip-77) [inline src/test/java/org/jalasoft/moi/model/exceptions/ParametersExceptionTest.java]: Missing additional space

File: src/test/java/org/jalasoft/moi/model/exceptions/ProcessIDExceptionTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        Executer executer = new Executer(new ProcessCache());
+        executer.execute(null);
+    }
+
+    @Test(expected = ProcessIDException.class)
+    public void throwsExceptionWhenCommandInvalidTest() throws CommandBuildException, ResultException, ProcessIDException {
+        Executer executer = new Executer(new ProcessCache());
+        executer.execute("wrong");
+    }
+}
```
Comment (Snip-77) [inline src/test/java/org/jalasoft/moi/model/exceptions/ProcessIDExceptionTest.java]: Missing additional space.

**Review by Snip-77 — CHANGES_REQUESTED** [review]
The handlers no longer exist. To we need tests for this?

**Review by Snip-77 — COMMENTED** [review]
Waiting for the other reviewers to aprove

**Review by carlosmc23 — DISMISSED** [review]

File: build.gradle [hunk 5]
```diff
@@ -14,18 +14,31 @@ repositories {
...
     implementation 'org.springframework.boot:spring-boot-dependencies:2.2.5.RELEASE'
     implementation 'org.springframework.boot:spring-boot-starter-web'
     implementation 'io.jsonwebtoken:jjwt:0.2'
     testImplementation('org.springframework.boot:spring-boot-starter-test') {
         exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
     }
+
+    testImplementation(
+            'junit:junit:4.12',
+            'org.junit.jupiter:junit-jupiter:5.4.2',
```
Comment (luqp) [inline build.gradle:31]: Are you using both libraries in your unit test?
I believe that only one is necessary

File: src/main/java/org/jalasoft/moi/controller/endpoints/FileController.java [hunk 6]
```diff
@@ -105,9 +112,24 @@ public void deleteFileById(@PathVariable Long id) {
      * @return the output from the execution
      */
     @PostMapping(path = "/execute")
-    public String executeCode(@RequestParam(value = "Project Id") Long projectID){
+    public String executeCode(@RequestParam(value = "Project Id") Long projectID) throws IOException {
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/endpoints/FileController.java]: It's an endpoint, I believe you should not throw an exception in the signature of the method.

File: src/main/java/org/jalasoft/moi/controller/endpoints/LanguageController.java [hunk 7]
```diff
@@ -44,7 +45,7 @@
      */
     @RequestMapping(method = RequestMethod.PUT, path = "/q")
     public String processInput(@RequestParam(value = "userInput") String userInput,
-                               @RequestParam(value = "pid") Long pid) {
+                               @RequestParam(value = "pid") Long pid) throws ResultException, InputParametersException {
         InputParameters answer = new Answer();
```
Comment (luqp) [inline src/main/java/org/jalasoft/moi/controller/endpoints/LanguageController.java:49]: Here the same case, I believe you should not throw an exception in the signature of the method.

**Review by carlosmc23 — APPROVED** [review]

**Review by Snip-77 — APPROVED** [review]

**Pull request merged**